src/asmcnc/comms/grbl_settings_manager.py

```diff
@@ -258,29 +257,33 @@ def on_persistent_setting(self, instance, value, setting):
             
     def on_console_specific_setting(self, instance, value, setting):
         """
-        Called when a console specific setting is read in or changed.
+        Called when a console specific setting is read in.
 
         Checks if the read in value differs from last recorded value.
         """
         descriptions = {
             51: "SC2 ($51) setting"
         }
 
-        # Check if this is the first read
-        if not self._has_read_51:
-            # If setting has not yet been saved then save it
-            if self._machine_saved_data[setting] == -1:
-                self._machine_saved_data[setting] = value
-                self.save_machine_data_to_file()
-                log('First startup after update? Setting {} saved to file: {}'.format(descriptions[setting], value))
-            # Otherwise if the value does not match then restore it
-            elif value != self._machine_saved_data[setting]:
-                self.machine.write_dollar_setting(setting, self._machine_saved_data[setting])
-                log('Restored {} from file: {}'.format(descriptions[setting], self._machine_saved_data[setting]))
-        # After startup, changes are intentional, so just overwrite internal value
+        # If setting has not yet been saved then save it
+        if self._machine_saved_data[setting] == -1:
+            self._machine_saved_data[setting] = value
+            self.save_machine_data_to_file()
+            log('First startup after update? Setting {} saved to file: {}'.format(descriptions[setting], value))
+        # Otherwise if the value does not match then restore it
         elif value != self._machine_saved_data[setting]:
+            self.machine.write_dollar_setting(setting, self._machine_saved_data[setting])
+            log('Restored {} from file: {}'.format(descriptions[setting], self._machine_saved_data[setting]))
+
+    def save_dollar_setting(self, setting, value):
```

Comment:
rename this to something like save_console_settings_to_file
It is not generally used for all dollar settings and it saves it to a file, not to the zh. Makes it clearer.